packages/product/src/services/product-category.ts

```diff
@@ -76,6 +77,17 @@ export default class ProductCategoryService<
     delete filters.include_descendants_tree
     delete filters.include_ancestors_tree
 
+    // Apply free text search filter
+    if (filters?.q) {
```

Comment:
These are temporary until we figure out what we want to do with the ProductCategoryService. Right now, it is a custom service not using the abstract service factory, so the free text search is not supported out of the box. 

cc @adrien2p